lib/phoenix/endpoint/supervisor.ex

```diff
@@ -11,17 +11,47 @@ defmodule Phoenix.Endpoint.Supervisor do
   Starts the endpoint supervision tree.
   """
   def start_link(otp_app, mod, opts \\ []) do
-    case Supervisor.start_link(__MODULE__, {otp_app, mod, opts}, name: mod) do
-      {:ok, _} = ok ->
-        warmup(mod)
-        log_access_info(otp_app, mod)
-        ok
-
+    with {:ok, _} <- check_force_ssl(otp_app, mod),
+         {:ok, _} = ok <- Supervisor.start_link(__MODULE__, {otp_app, mod, opts}, name: mod) do
+      warmup(mod)
+      log_access_info(otp_app, mod)
+      ok
+    else
       {:error, _} = error ->
         error
     end
   end
 
+  defp check_force_ssl(otp_app, mod) do
+    compile_config = mod.__compile_config__() |> Keyword.get(:force_ssl)
```

Comment:
Instead of validating only `force_ssl`, we can simply validate all keys in `compile_config`. That was the goal with making it generic. In a nutshell, you can traverse this keyword list, comparing it with the values in runtime_config.

Reply:
Oh I see, sorry about that. I wasn't sure if the error message would be generic enough for future keys, but I think it should be alright. Updated.